src/App.tsx

```diff
@@ -1,29 +1,68 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+type PerPage = '3' | '5' | '10' | '20';
+
+const three: PerPage = '3';
+const five: PerPage = '5';
+const ten: PerPage = '10';
+const twenty: PerPage = '20';
+
+const totalItems: number = 42;
+
+interface Item {
+  id: number;
+  title: string;
+}
+
+const getEndItem = (
+  perPage: number,
+  currentPage: number,
+  amountItems: number,
+): number => {
+  if (perPage * currentPage > amountItems) {
+    return amountItems;
+  }
+
+  return perPage * currentPage;
+};
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState<PerPage>(five);
+  const startItem = +perPage * currentPage - +perPage + 1;
+  const endItem: number = getEndItem(+perPage, currentPage, totalItems);
+  const items: Item[] = getNumbers(startItem, endItem).map(n => {
+    return { id: n, title: `Item ${n}` };
+  });
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {startItem} - {endItem} of 42)
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            onChange={event => {
+              setPerPage(event.target.value as PerPage);
+              setCurrentPage(1);
+            }}
+          >
+            <option value={three}>3</option>
+            <option value={five} selected>
+              5
+            </option>
```

Comment:
The 'selected' attribute is not used in React. Instead, you should set the default value of the <select> element using the 'value' attribute on the <select> tag itself and manage the selected option through the component's state.